dinstaller-cli/src/main.rs

```diff
@@ -27,91 +29,88 @@ struct Cli {
     pub format: Option<Format>,
 }
 
-async fn probe(manager: &ManagerClient<'_>) {
-    let probe = task::spawn(async {
-        // use new manager here
-        let another_manager = ManagerClient::new(dinstaller_lib::connection().await.unwrap())
-            .await
-            .unwrap();
-        another_manager.probe().await.unwrap()
-    });
-    block_on(show_progress(&manager));
+async fn probe(manager: &ManagerClient<'_>) -> Result<(), Box<dyn Error>> {
+    let another_manager = build_manager().await?;
+    let probe = task::spawn(async move { another_manager.probe().await });
```

Comment:
do we need that async/await here? would't be enough to have just `move { another_manager.probe() }` ?

Reply:
I think so, [task:spawn](https://docs.rs/async-std/latest/async_std/task/fn.spawn.html) requires a `Future`.

Reply:
yeap and probe returns Future, not?

Reply:
Yes, I guess it might work. But you cannot `await` for it if you are not in an async context (and the future is lazy, so it would not run). I tried your suggestion, and it does not work:

```
   Compiling dinstaller-cli v0.1.0 (/home/imobach/Proyectos/yast/repos/dinstaller-rs/dinstaller-cli)
error[E0277]: `[closure@dinstaller-cli/src/main.rs:34:29: 34:36]` is not a future
  --> dinstaller-cli/src/main.rs:34:29
   |
34 |     let probe = task::spawn(move || another_manager.probe());
   |                 ----------- ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ `[closure@dinstaller-cli/src/main.rs:34:29: 34:36]` is not a future
   |                 |
   |                 required by a bound introduced by this call
   |
   = help: the trait `Future` is not implemented for closure `[closure@dinstaller-cli/src/main.rs:34:29: 34:36]`
   = note: [closure@dinstaller-cli/src/main.rs:34:29: 34:36] must be a future or must implement `IntoFuture` to be awaited
```